File: app/src/main/java/com/jamieadkins/gwent/deck/list/DeckListContract.java
package com.jamieadkins.gwent.deck.list;

import androidx.annotation.NonNull;

import com.jamieadkins.gwent.card.list.CardsContract;
import com.jamieadkins.gwent.domain.deck.model.GwentDeck;

import java.util.List;

/**
 * Specifies the contract between the view and the presenter.
 */

public interface DeckListContract {
    interface View {
        void showDecks(@NonNull List<GwentDeck> decks);

        void showDeckDetails(@NonNull String deckId);

        void showLoadingIndicator(Boolean loading);
    }

    interface Presenter extends CardsContract.Presenter {
        // Empty.
    }
}

File: app/src/main/java/com/jamieadkins/gwent/deck/list/DeckListContract.kt
package com.jamieadkins.gwent.deck.list

import com.jamieadkins.gwent.domain.deck.model.GwentDeck
import com.jamieadkins.gwent.base.MvpPresenter

interface DeckListContract {

    interface View {

        fun showDecks(decks: List<GwentDeck>)

        fun showDeckDetails(deckId: String)

        fun showLoadingIndicator(loading: Boolean)
    }

    interface Presenter : MvpPresenter
}
